### ISSUE TITLE
Brew CLI app is not on Homebrew anymore

### ISSUE THREAD

**Issue by chmac** [laurent22/joplin#5281]
The same info is repeated in the docs in these 2 places:

- https://joplinapp.org/help/#unsupported-methods
- https://joplinapp.org/terminal/#unsupported-methods

> macOS | brew install joplin

This is mistaken. The brew cask installs **only** the desktop version.

The cask data is [here](https://formulae.brew.sh/api/cask/joplin.json) and it specifies that only `Joplin-x.x.x.dmg` is downloaded and installed. No terminal part.

**Comment by laurent22** [comment]
As you found out those are unsupported methods so we don't know what they do. If the info is incorrect, feel free to create a pull request to correct the issue.

**Comment by chmac** [comment]
@laurent22 The brew method can simply be removed from there as it is not related to the terminal app. I tried to figure out where these docs were located, but I found HTMl files in the `docs/` directory. Should I update those files in the PR, or are those files generated from some other docs stored elsewhere? Happy to submit a PR removing that row if somebody can point me in the direction of where to make the changes...

PS> Thanks a lot for Joplin, it's really an excellent evernote replacement. ❤️

**Comment by tessus** [comment]
Please note: `brew install joplin` installed the terminal version. They changed the formula and now it doesn't. So the documentation was correct when it was written.

I'll remove the entry from the documentation - it's not supported anyway. And personally I'm not a fan of brew either.

**Comment by tessus** [comment]
https://github.com/laurent22/joplin/commit/6a17f717cf58214e226229ef5e38bb3f6151c9ef

**Comment by laurent22** [comment]
I wonder why they've changed it anyway. It seems it would make more sense to install the terminal app from a terminal command. And they could have a separate `brew install joplin-desktop` for the desktop app.

**Comment by tessus** [comment]
Well, cask is usually precompiled stuff and is invoked with --cask, but I have no idea why they did what they did.

On the other hand, we might want to move the unsupported methods section one up, because it is not necessarily part of the terminal app. The aur package installs terminal and desktop and there are a few others we could add. The snap and the rpm package. Not sure if there's already a deb package.

**Comment by chmac** [comment]
@laurent22 The formulae is a "Cask", which in brew parlance means it installs a desktop app. It would be possible to add a formulae that installs the terminal application, but as the terminal app is shipped via npm anyway, it's maybe not worth it. The only advantage would be that the `joplin-terminal` formulae could depend on `node` and so users would not need to install node first. But as brew formulae are all in ruby, it's outside of my expertise, otherwise I'd give it a bash trying to create a formulae for the terminal app...

**Comment by laurent22** [comment]
I think the main advantage of Homebrew for the CLI app is that it would be easy to update using `brew upgrade` like any other app. On the other hand the desktop app already has auto-update so Homebrew is unnecessary for it.

I feel they've done things upside down a bit. I don't use Homebrew for the CLI app but only because I also develop it. As a user I would much prefer Homebrew for it.

**Comment by chmac** [comment]
@laurent22 Makes sense. I tried creating a new homebrew formula for joplin, but I hit a problem with the tests. When I run `joplin version` during the brew test, I get a keychain pop up. I believe [brew tests must be non interactive](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula). I can't find any option to disable the keychain service as a command line flag, so I'm not sure how to workaround this. I've created a [draft PR on the homebrew repo](https://github.com/Homebrew/homebrew-core/pull/83099) to solicit feedback there also.

**Comment by laurent22** [comment]
That's because it needs the keychain to save secure settings like passwords, and I think there's some code that initialize the keychain on first run.

Could you find out if your pull request has any chance of being accepted? If so I'm happy to help with getting the formula working, for example by adding a new `joplin test` command that could be used for Homebrew (and other cases) and that would skip the keychain check.

**Comment by carlocab** [comment]
Hi there, Homebrew maintainer here.

The Homebrew `joplin` formula was removed in https://github.com/Homebrew/homebrew-core/pull/59053 in favour of the cask because we couldn't test that it was working in CI -- presumably, this is because of the Keychain issue referenced above.

If you add a command that provides a quick smoke test that the `joplin` install isn't completely broken, I'd be in favour of adding it back to Homebrew/core. I don't see why other maintainers would be opposed either.

**Comment by chmac** [comment]
I've added a feature request to discourse https://discourse.joplinapp.org/t/add-a-disable-keychain-command-line-flag/19394

Hopefully we can find a way to disable the keychain services (either by command line flag, env var, or otherwise) and then we can get joplin back into brew. 🚀

**Comment by laurent22** [comment]
Ok that's great to know, thanks for confirming @carlocab.

**Comment by laurent22** [comment]
Oh just noticed I was pinged on the initial issue that removed the CLI app but I missed it, sorry about that.

**Comment by chmac** [comment]
@laurent22 I've spent now >8h trying to get the brew formula to work. The formula works on macOS intel machines but fails the tests on arm64 (macOS M1 chips) and on ubuntu. I'm not familiar enough with brew to know how this stuff works, and personally I've found the brew docs to be rather hard to follow. I'm personally dropping out of the brew effort now, hopefully somebody else can pick this up and reach success with it. Best of luck.
